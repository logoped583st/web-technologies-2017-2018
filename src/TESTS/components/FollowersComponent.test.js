import React from 'react';
import FollowersCompomponent from '../../components/FollowersComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import { HashRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('Followers component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        reducerUser: {
            login: 'logoped583st'
        },
        reducerFollowers: [
            {
                login: 'logoped583st',
                avatar_url: 'https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4'
            }
        ]
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('Followers component rendered succesfully', () => {
        const userFollowers = mount(
            <HashRouter>
                <FollowersCompomponent store={store} />
            </HashRouter>
        );
        expect(mountToJson(userFollowers)).toMatchSnapshot();
    });

    it('Followers rendered without user login', () => {
        const fakeInitialState = {
            reducerUser: {
                login: ''
            },
            reducerFollowers: [
                {
                    login: 'logoped583st',
                    avatar_url: 'https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4'
                }
            ]
        }
        const fakeStore = mockStore(fakeInitialState);
        const userFollowers = mount(
            <HashRouter>
                <FollowersCompomponent store={fakeStore} />
            </HashRouter>
        );
        expect(userFollowers.prop('login')).toBe(undefined);
        expect(userFollowers.prop('follower')).toBe(undefined);

        expect(mountToJson(userFollowers)).toMatchSnapshot();
    })

    it('UserFollowers rendered without user login', () => {
        const fakeInitialState = {
            reducerUser: {
                login: ''
            },
            reducerFollowers: [
                {
                    login: 'logoped583st',
                    avatar_url: 'https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4'
                }
            ]
        }
        const fakeStore = mockStore(fakeInitialState);
        const userFollowers = mount(
            <HashRouter>
                <FollowersCompomponent store={fakeStore} />
            </HashRouter>
        );
        expect(userFollowers.prop('login')).toBe(undefined);
        expect(userFollowers.prop('followers')).toBe(undefined);


        expect(mountToJson(userFollowers)).toMatchSnapshot();
    })

    it('UserFollowers rendered without user login', () => {
        const fakeInitialState = {
            reducerUser: {
                login: ''
            },
            reducerFollowers: [
                {
                    login: '',
                    avatar_url: 'https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4'
                }
            ]
        }
        const fakeStore = mockStore(fakeInitialState);
        const userFollowers = mount(
            <HashRouter>
                <FollowersCompomponent store={fakeStore} />
            </HashRouter>
        );

        expect(userFollowers.prop('followers')).toBe(undefined);
        expect(mountToJson(userFollowers)).toMatchSnapshot();
    })

});