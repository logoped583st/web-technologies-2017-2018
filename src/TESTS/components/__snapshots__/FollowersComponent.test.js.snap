// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Followers component test Followers component rendered succesfully 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(FollowersComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <FollowersComponent
        fetchFollowersRequest={[Function]}
        login="logoped583st"
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      />
    </Connect(FollowersComponent)>
  </Router>
</HashRouter>
`;

exports[`Followers component test Followers rendered without user login 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(FollowersComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <FollowersComponent
        fetchFollowersRequest={[Function]}
        login=""
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      />
    </Connect(FollowersComponent)>
  </Router>
</HashRouter>
`;

exports[`Followers component test UserFollowers rendered without user login 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(FollowersComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <FollowersComponent
        fetchFollowersRequest={[Function]}
        login=""
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      />
    </Connect(FollowersComponent)>
  </Router>
</HashRouter>
`;

exports[`Followers component test UserFollowers rendered without user login 2`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(FollowersComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <FollowersComponent
        fetchFollowersRequest={[Function]}
        login=""
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      />
    </Connect(FollowersComponent)>
  </Router>
</HashRouter>
`;
